drivers/espi/espi_it8xxx2.c

```diff
@@ -0,0 +1,1260 @@
+		slave_reg->ESGCTRL2 &= ~IT8XXX2_ESPI_INPUT_PAD_GATING;
+	} else {
+		/* Disable eSPI pad. */
+		slave_reg->ESGCTRL2 |= IT8XXX2_ESPI_INPUT_PAD_GATING;
+	}
+}
+
+void espi_it8xxx2_espi_reset_isr(const struct device *port,
+				struct gpio_callback *cb, uint32_t pins)
+{
+	struct espi_it8xxx2_data *const data = DRV_DATA(ESPI_IT8XXX2_SOC_DEV);
+	struct espi_event evt = {ESPI_BUS_RESET, 0, 0};
+	bool espi_reset = gpio_pin_get(port, (find_msb_set(pins) - 1));
+
+	if (!(espi_reset)) {
+		/* Reset vwidx_cached_flag[] when espi_reset# asserted. */
+		espi_it8xxx2_reset_vwidx_cache(ESPI_IT8XXX2_SOC_DEV);
+	}
+
+	evt.evt_data = espi_reset;
+	espi_send_callbacks(&data->callbacks, ESPI_IT8XXX2_SOC_DEV, evt);
+
+	LOG_INF("eSPI reset %sasserted", espi_reset ? "de" : "");
+}
+
+/* eSPI reset# is enabled on GPD2 */
+#define ESPI_IT8XXX2_ESPI_RESET_PORT DEVICE_DT_GET(DT_NODELABEL(gpiod))
+#define ESPI_IT8XXX2_ESPI_RESET_PIN  2
+static void espi_it8xxx2_enable_reset(void)
+{
+	static struct gpio_callback espi_reset_cb;
+
+	/* eSPI reset is enabled on GPD2 */
+	IT8XXX2_GPIO_GCR = (IT8XXX2_GPIO_GCR & ~0x6) |
+		(IT8XXX2_GPIO_GCR_ESPI_RST_D2 << IT8XXX2_GPIO_GCR_ESPI_RST_POS);
+	/* enable eSPI reset isr */
+	gpio_init_callback(&espi_reset_cb, espi_it8xxx2_espi_reset_isr,
+				BIT(ESPI_IT8XXX2_ESPI_RESET_PIN));
+	gpio_add_callback(ESPI_IT8XXX2_ESPI_RESET_PORT, &espi_reset_cb);
+	gpio_pin_interrupt_configure(ESPI_IT8XXX2_ESPI_RESET_PORT,
+					ESPI_IT8XXX2_ESPI_RESET_PIN,
+					GPIO_INT_TRIG_BOTH);
+}
+
+static struct espi_it8xxx2_data espi_it8xxx2_data_0;
+static const struct espi_it8xxx2_config espi_it8xxx2_config_0 = {
+	.base_espi_slave = DT_INST_REG_ADDR_BY_IDX(0, 0),
+	.base_espi_vw = DT_INST_REG_ADDR_BY_IDX(0, 1),
+	.base_espi_queue1 = DT_INST_REG_ADDR_BY_IDX(0, 2),
+	.base_espi_queue0 = DT_INST_REG_ADDR_BY_IDX(0, 3),
+	.base_ec2i = DT_INST_REG_ADDR_BY_IDX(0, 4),
+	.base_kbc = DT_INST_REG_ADDR_BY_IDX(0, 5),
+	.base_pmc = DT_INST_REG_ADDR_BY_IDX(0, 6),
+};
+
+DEVICE_DT_INST_DEFINE(0, &espi_it8xxx2_init, NULL,
+		    &espi_it8xxx2_data_0, &espi_it8xxx2_config_0,
+		    PRE_KERNEL_2, CONFIG_ESPI_INIT_PRIORITY,
+		    &espi_it8xxx2_driver_api);
+
+static int espi_it8xxx2_init(const struct device *dev)
+{
+	const struct espi_it8xxx2_config *const config = DRV_CONFIG(dev);
+	struct espi_vw_regs *const vw_reg =
+		(struct espi_vw_regs *)config->base_espi_vw;
+	struct espi_slave_regs *const slave_reg =
+		(struct espi_slave_regs *)config->base_espi_slave;
+	struct gctrl_it8xxx2_regs *const gctrl = ESPI_IT8XXX2_GET_GCTRL_BASE;
+
+	/*
+	 * bit[7-6] = 1: The VCC power status is treated as power-on.
+	 * The VCC supply of eSPI and related functions (EC2I, KBC, PMC and
+	 * PECI). It means VCC should be logic high before using these
+	 * functions, or firmware treats VCC logic high.
+	 *
+	 * bit[3] = 0: The reset source of PNPCFG is RSTPNP bit in RSTCH
+	 * register and WRST#.
+	 *
+	 * bit[2] = 1: Enable global reset.
+	 */
+	gctrl->GCTRL_RSTS = 0x44;
```

Comment:
Prefer to use macros to avoid long comments. It also simplifies usage if need same bit usage elsewhere

Reply:
Done.